core/editor/src/tools/mod.rs

```diff
@@ -1,14 +1,30 @@
-use crate::events::{ModKeys, MouseState, TracePoint};
-use crate::{events::Trace, Color};
+mod crop;
+mod ellipse;
+mod line;
+mod navigate;
+mod path;
+mod pen;
+mod rectangle;
+mod sample;
+mod select;
+mod shape;
+
+use crate::events::{Event, ModKeys, MouseState, Trace, TracePoint};
+use crate::Color;
 use std::collections::HashMap;
 
+pub trait Tool {
+	fn handle_input(&mut self, event: Event);
```

Comment:
Didn't you want to rename that function to `handle`at some point?

Reply:
No, I actually wanted to rename it to `handle_input` and I did (you suggested `handle`).

Reply:
Oh, sorry I got confused, I originally used the name `handle_input` and just abbreviated it from there on, so we are on the same page